pkg/lso/localvolumediscovery_test.go

```diff
@@ -0,0 +1,404 @@
+			namespace:     "",
+			expectedError: "localVolumeDiscovery 'nsname' cannot be empty",
+			client:        true,
+		},
+		{
+			name:          defaultLocalVolumeDiscoveryName,
+			namespace:     defaultLocalVolumeDiscoveryNamespace,
+			expectedError: "",
+			client:        false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var testSettings *clients.Settings
+		if testCase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{})
+		}
+
+		testLocalVolumeDiscovery := NewLocalVolumeDiscoveryBuilder(testSettings, testCase.name, testCase.namespace)
+
+		if testCase.expectedError == "" {
+			if testCase.client {
+				assert.Equal(t, testCase.name, testLocalVolumeDiscovery.Definition.Name)
+				assert.Equal(t, testCase.namespace, testLocalVolumeDiscovery.Definition.Namespace)
+			} else {
+				assert.Nil(t, testLocalVolumeDiscovery)
+			}
+		} else {
+			assert.Equal(t, testCase.expectedError, testLocalVolumeDiscovery.errorMsg)
+			assert.NotNil(t, testLocalVolumeDiscovery.Definition)
+		}
+	}
+}
+
+func TestLocalVolumeDiscoveryExists(t *testing.T) {
+	testCases := []struct {
+		testLocalVolumeDiscovery *LocalVolumeDiscoveryBuilder
+		expectedStatus           bool
+	}{
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(buildLVDClientWithDummyObject()),
+			expectedStatus:           true,
+		},
+		{
+			testLocalVolumeDiscovery: buildInValidLVDObjectBuilder(buildLVDClientWithDummyObject()),
+			expectedStatus:           false,
+		},
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedStatus:           false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		exist := testCase.testLocalVolumeDiscovery.Exists()
+		assert.Equal(t, testCase.expectedStatus, exist)
+	}
+}
+
+func TestLocalVolumeDiscoveryGet(t *testing.T) {
+	testCases := []struct {
+		testLocalVolumeDiscovery *LocalVolumeDiscoveryBuilder
+		expectedError            error
+	}{
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(buildLVDClientWithDummyObject()),
+			expectedError:            nil,
+		},
+		{
+			testLocalVolumeDiscovery: buildInValidLVDObjectBuilder(buildLVDClientWithDummyObject()),
+			expectedError:            fmt.Errorf("localVolumeDiscovery 'name' cannot be empty"),
+		},
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError: fmt.Errorf("localvolumediscoveries.local.storage.openshift.io " +
+				"\"auto-discover-devices\" not found"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		localVolumeDiscoveryObj, err := testCase.testLocalVolumeDiscovery.Get()
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, localVolumeDiscoveryObj.Name, testCase.testLocalVolumeDiscovery.Definition.Name)
+			assert.Equal(t, localVolumeDiscoveryObj.Namespace, testCase.testLocalVolumeDiscovery.Definition.Namespace)
+			assert.Nil(t, err)
```

Comment:
Check error unconditionally

Reply:
@trewest I cannot check it unconditionally in this case, I need an access to the error message to compare and when there is no error I'm failing on panic: runtime error: invalid memory address or nil pointer dereference

Reply:
This assertion doesn't work?
```
assert.Equal(t, testCase.expectedError, err)
```